src/ToolingSharedLibrary/Includes/VbsEnclaveABI/Shared/ConversionHelpers.h

```diff
@@ -0,0 +1,446 @@
+        }
+    }
+
+    template <UniquePtr Target, typename Src>
+    inline std::decay_t<Target> ConvertToUniquePtr(const Src& src)
+    {
+        using TargetInnerType = unique_ptr_inner_type_t<Target>;
+
+        auto conversion_func = [] (auto&& value)
+        {
+            using SrcInnerType = std::decay_t<decltype(value)>;
+            auto ptr = std::make_unique<TargetInnerType>();
+            *ptr = ConvertType<TargetInnerType>(value);
+            return ptr;
+        };
+
+        return ConvertToWrapper<std::unique_ptr, TargetInnerType>(src, conversion_func);
+    }
+
+    template <Optional Target, typename Src>
+    inline std::decay_t<Target> ConvertToOptional(const Src& src)
+    {
+        using TargetInnerType = optional_inner_type_t<Target>;
+
+        auto conversion_func = [] (auto&& value)
+        {
+            using SrcInnerType = std::decay_t<decltype(value)>;
+            return std::make_optional<TargetInnerType>(ConvertType<TargetInnerType>(value));
+        };
+
+        return ConvertToWrapper<std::optional, TargetInnerType>(src, conversion_func);
+    }
+
+    template<typename T>
+    inline std::wstring ConvertToStdWString(const T& wstr)
+    {
+        if constexpr (UniquePtr<T>)
+        {
+            if (!wstr)
+            {
+                return {};
+            }
+
+            return std::wstring(wstr->wchars.begin(), wstr->wchars.end());
+        }
+        else
+        {
+            return std::wstring(wstr.wchars.begin(), wstr.wchars.end());
+        }
+    }
+
+    template<typename Target>
+    inline Target CreateWStringT(const std::wstring& wchars)
+    {
+        using DecayedType = std::decay_t<Target>;
+        if constexpr (UniquePtr<DecayedType>)
+        {
+            using InnerType = unique_ptr_inner_type_t<DecayedType>;
+            auto ptr = std::make_unique<InnerType>();
+            ptr->wchars.assign(wchars.begin(), wchars.end());
+            return ptr;
+        }
+        else if constexpr (NonWrappedStruct<DecayedType>)
+        {
+            DecayedType wcharT {};
+            wcharT.wchars.assign(wchars.begin(), wchars.end());
+            return wcharT;
+        }
+        else
+        {
+            static_assert(always_false<Target>::value, "CreateWStringT: Unsupported Target (must be unique_ptr or struct)");
+        }
+    }
+
+    template <typename TargetVector, typename SrcContainer, typename ConvertFunc>
+    TargetVector TransformVector(const SrcContainer& src, ConvertFunc&& conversion_func)
+    {
+        TargetVector target_vector;
+        target_vector.reserve(src.size());
+        for (const auto& value : src)
+        {
+            target_vector.emplace_back(conversion_func(value));
+        }
+
+        return target_vector;
+    }
+
+    template <typename StdArray, typename SrcVector, typename ConvertFunc>
+    StdArray TransformVectorToStdArray(const SrcVector& src, ConvertFunc&& conversion_func)
```

Comment:
>TransformVectorToStdArray [](http://example.com/codeflow?start=13&length=25)

nit: imo, just call this TransformToStdArray - who cares if it's a vector - any std container that supports iterators would work